vulnerabilities/importers/debian.py

```diff
@@ -21,115 +21,194 @@
 #  VulnerableCode is a free software tool from nexB Inc. and others.
 #  Visit https://github.com/nexB/vulnerablecode/ for support and download.
 
-import dataclasses
+import logging
 from typing import Any
+from typing import Iterable
 from typing import List
 from typing import Mapping
-from typing import Set
 
 import requests
-from dateutil import parser as dateparser
+from django.db.models.query import QuerySet
 from packageurl import PackageURL
+from univers.version_range import DebianVersionRange
+from univers.versions import DebianVersion
 
+from vulnerabilities.helpers import AffectedPackage as LegacyAffectedPackage
+from vulnerabilities.helpers import get_affected_packages_by_patched_package
+from vulnerabilities.helpers import get_item
 from vulnerabilities.helpers import nearest_patched_package
 from vulnerabilities.importer import AdvisoryData
+from vulnerabilities.importer import AffectedPackage
 from vulnerabilities.importer import Importer
 from vulnerabilities.importer import Reference
+from vulnerabilities.importer import UnMergeablePackageError
+from vulnerabilities.improver import MAX_CONFIDENCE
+from vulnerabilities.improver import Improver
+from vulnerabilities.improver import Inference
+from vulnerabilities.models import Advisory
 
+logger = logging.getLogger(__name__)
 
-class DebianImporter(Importer):
-    def __enter__(self):
-        if self.response_is_new():
-            self._api_response = self._fetch()
+IGNORABLE_VERSIONS = {"3.8.20-4."}
 
-        else:
-            self._api_response = {}
 
-    def updated_advisories(self) -> Set[AdvisoryData]:
-        advisories = []
+class DebianImporter(Importer):
 
-        for pkg_name, records in self._api_response.items():
-            advisories.extend(self._parse(pkg_name, records))
+    spdx_license_expression = "MIT"
+    license_url = "https://www.debian.org/license"
 
-        return self.batch_advisories(advisories)
+    api_url = "https://security-tracker.debian.org/tracker/data/json"
 
-    def _fetch(self) -> Mapping[str, Any]:
-        return requests.get(self.config.debian_tracker_url).json()
+    def get_response(self):
```

Comment:
IMHO this really ought to be a shared helper so we can better configure it with proxy and more. This can be done later though